lib/utils/log/slog.go

```diff
@@ -68,6 +119,47 @@ func addTracingContextToRecord(ctx context.Context, r *slog.Record) {
 	}
 }
 
+var defaultFormatFields = []string{LevelField, ComponentField, CallerField, TimestampField}
+
+var knownFormatFields = map[string]struct{}{
+	LevelField:     {},
+	ComponentField: {},
+	CallerField:    {},
+	TimestampField: {},
+}
+
+// ValidateFields ensures the provided fields map to the allowed fields. An error
+// is returned if any of the fields are invalid.
+func ValidateFields(formatInput []string) (result []string, err error) {
```

Comment:
Could this be made private and made a responsibility of Initialize? Perhaps after the duplicated code on integrations/ is removed?